### DISCUSSION TITLE
Does the nerdctl support the external snapshotter plugin?

### DISCUSSION THREAD

**Discussion by xinnywinne** [containerd/nerdctl#2900] [Q&A]
We have created an external snapshotter plugin named "csfs" and we would like to use it as the default snapshotter. We modified the containerd configuration file under `[plugins] [plugins."io.containerd.grpc.v1.cri".containerd]` to use this external snapshotter plugin. It works perfectly with the `crictl` command, but not with the `nerdctl` command.

Here is the command I am using: `nerdctl --address /run/containerd/containerd.sock --snapshotter=csfs pull docker.io/library/alpine:3.18`. It seems that instead of using the "csfs" snapshotter, `nerdctl` might still be using the overlayfs snapshotter.

So, I am wondering if `nerdctl` actually supports the custom-defined external snapshotter plugin or not.

**Comment by AkihiroSuda** [comment]
`nerdctl --snapshotter=SNAPSHOTTER` should just work.

For external plugins you need to have `[proxy_plugins]` in `config.toml`
https://github.com/containerd/containerd/blob/main/docs/PLUGINS.md#configuration
